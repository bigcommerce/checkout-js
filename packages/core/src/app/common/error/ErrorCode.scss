@import '../../ui/Base';

.errorCode {
    font-size: fontSize("tiny");
    opacity: 0.4;

    .errorCode-value {
        word-break: break-word;
    }
}
