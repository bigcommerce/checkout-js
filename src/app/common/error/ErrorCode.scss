@import '../../../../node_modules/@bigcommerce/citadel/src/settings/global/global';

.errorCode {
    font-size: fontSize("tiny");
    opacity: 0.4;

    .errorCode-value {
        word-break: break-word;
    }
}
