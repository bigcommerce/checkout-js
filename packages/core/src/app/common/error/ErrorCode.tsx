import React, { FunctionComponent, memo, ReactNode } from 'react';

import { TranslatedString } from '../../locale';

import './ErrorCode.scss';

export interface ErrorCodeProps {
    code: string;
    label?: ReactNode;
}

const ErrorCode: FunctionComponent<ErrorCodeProps> = ({ code, label }) => {
    return (
        <div className="errorCode">
            <span className="errorCode-label">
                {label ?? <TranslatedString id="common.error_code" />}
            </span>{' '}
            <span className="errorCode-value">{code}</span>
        </div>
    );
};

export default memo(ErrorCode);
