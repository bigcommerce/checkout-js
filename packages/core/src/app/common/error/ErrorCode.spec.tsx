import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import ErrorCode from './ErrorCode';

describe('ErrorCode Component', () => {
    it('renders error code component with provided code', () => {
        const tree = shallow(<ErrorCode code="foo" />);

        expect(toJson(tree)).toMatchSnapshot();
    });
});
