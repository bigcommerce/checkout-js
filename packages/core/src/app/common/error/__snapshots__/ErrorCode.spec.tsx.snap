// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorCode Component renders error code component with provided code 1`] = `
<div
  className="errorCode"
>
  <span
    className="errorCode-label"
  >
    <WithLanguage(TranslatedString)
      id="common.error_code"
    />
  </span>
   
  <span
    className="errorCode-value"
  >
    foo
  </span>
</div>
`;
